### PR TITLE
Autogenerate `OWNERS_ALIASES`

### PR DESCRIPTION
**Pull request by upodroid**
Part of https://github.com/kubernetes/release/issues/2175

/cc @saschagrunert @puerco 

1st part of the automated process, generating OWNER_ALIASES here.

2nd part will be a periodic prowjob that propagates the OWNER_ALIASES to all repos.

### CODE REVIEW

File: hack/verify-alias-gen.sh
Comment (saschagrunert) [inline hack/verify-alias-gen.sh:3]:
```suggestion
# Copyright 2022 The Kubernetes Authors.
```

**Review by saschagrunert — APPROVED** [review]
This is really cool! Let's wait for the owners if the location of the test files in `config` is fine for them.

**Comment by upodroid** [comment]
@ameukam Any thoughts? 

There will be one new change when humans are making membership changes. You'll need to run `make config` to autogenerate `*-OWNER_ALIASES` files when peribolos config is changed. There is also a new verify script that checks all PRs have the latest generated `OWNER_ALIASES`.

File: hack/verify-alias-gen.sh
Comment (upodroid) [reply hack/verify-alias-gen.sh]: Fixed

**Review by mrbobbytables — COMMENTED** [review]
I appreciate the effort, but this really should have been brought up with the github admins first. =/ We don't want to force syncing. Theres only a few repos where we would want to sync sigs.yaml changes and those are only in the kubernetes org, the place where we can guarantee that the members exist in that org in the first place.

Happy to continue to discuss, but that should include the other github admins as well.

**Comment by mrbobbytables** [comment]
/hold
To resolve convos in https://github.com/kubernetes/release/issues/2175 and

**Comment by upodroid** [comment]
@mrbobbytables I have resolved the comments in the issue:
- The automation is opt-in.
- The `OWNERS_ALIASES` are generated and calculated in `_output/` like the `make deploy` target.
- I haven't enrolled any repositories yet.

**Comment by ameukam** [comment]
@upodroid I think this PR should be moved to https://github.com/kubernetes/release. The main intent described in https://github.com/kubernetes/release/issues/2175 only concerns SIG Release Github repos. Can you please migrate the PR ?

**Comment by nikhita** [comment]
@upodroid thanks for the PR!

> Current OWNERS_ALIASES must be based on GitHub teams instead of manually created aliases.

We don't enforce any mapping between OWNERS_ALIASES and GitHub teams so this prerequisite won't work.

A better place to do this would be the maintainers tool - https://github.com/kubernetes-sigs/maintainers (see the `maintainers audit` command).

As next steps, if you are still interested in continuing this work, can you create an issue in this repo to continue the discussion?

**Comment by mrbobbytables** [comment]
I am overall +1 to GitHub team references. It covers multiple use-cases and would have less noise.

**Comment by mrbobbytables** [comment]
Going to go ahead and close this out as discussion has moved on.

/close

**Pull request closed without merging**
